src/config_rc.ts

```diff
@@ -0,0 +1,115 @@
+import * as Controller from './controller'
+import * as Util from './utils'
+import * as Config from './config'
+import * as Native from './native_background'
+import * as Messaging from './messaging'
+import DEFAULTS from './config_defaults'
+import Logger from './logging'
+
+const RC_NAME = 'rc-text'
+const RC_LOC_NAME = 'rc-location'
+const RC_AUTOLOAD_FS_NAME = 'rc-autoload'
+const RC_DEFAULT = `
+# This is the default RC file.
+# Because I am a nice guy, here are some commented settings for you.
+
+# Wipe addon settings every time this file is loaded
+# resetConfigToDefault
+
+# Use dark mode
+# set theme dark
+
+# Disable on reddit because RES
+# autocmd DocStart reddit.com mode ignore
+
+# Bind i to ':mode ignore'
+# bind i mode ignore
+
+# Access this page with ':st'
+# command st settings`
+
+const FSRC_NOT_FOUND = `
+# Something went wrong with loading the file from the filesystem. Please make
+# sure that all of the following are true:
+#  - You have a file at XDG_CONFIG_HOME/tridactyl/tridactylrc
+#  - You have installed the native app
+#  - No unusual errors are being reported in the console
+# If you are still lost, file a bug on GitHub or contact us on Gitter/Matrix. `
+
+export type RcLocation = 'filesystem' | 'browser'
+
+export async function initConfigFromRc(): Promise<void> {
+    await detectAndSetDefaultBrowserRc()
+
+    let isAutoload = await getAutoload()
+    let location = await getLocation()
+    let rcText: string
+
+    // Default value
+    if (!location) location = 'browser'
+    if (!isAutoload) isAutoload = false
+
+    console.info(`Loading RC from:`, location)
+
+    switch (location) {
+        case 'filesystem':
+            rcText = await getFilesystemRc()
+            break
+        case 'browser':
+            rcText = await getBrowserRc()
+            break
+        default:
+            throw `Location '${location}' is not valid. Please contact a dev
+            and tell them to use either 'filesystem' or 'browser'.`
+    }
+
+    // If we are autoloading and we actually loaded something from
+    // the filesystem, then set the rc as the thing
+    if (isAutoload && location === 'filesystem') setBrowserRc(rcText)
```

Comment:
This doesn't seem to work for me. Manually clicking load from file system and then run worked, but the tickbox below doesn't seem to do anything (tested on a fresh profile by loading Tridactyl as a temporary addon, ticking box, restarting Firefox, loading Tridactyl again. The tickbox is preserved but settings are not loaded (test binds are not bound).)

Reply:
Also, could this be turned on by default? I think really we want to load from an RC file if present unless told otherwise, so that the minimum amount of faff is required to set up a new computer with Tridactyl and all of your settings.